org.eclipse.winery.repository/src/main/java/org/eclipse/winery/repository/export/CSARExporter.java

```diff
@@ -191,6 +194,187 @@ public void writeCSAR(TOSCAComponentId entryId, OutputStream out) throws Archive
 	}
 
 	/**
+	 *
+	 * @param zos Output stream for the archive that should contain the file
+	 * @param ref Reference to the file that should be added to the archive
+	 * @param archivePath Path to the file inside the archive
+	 * @throws IOException thrown when the temporary directory can not be created
+	 */
+	private void addArtifactTemplateToZipFile(ArchiveOutputStream zos, RepositoryFileReference ref, String archivePath) throws IOException {
+		Utils.GitInfo gitInfo = Utils.getGitInformation((ArtifactTemplateDirectoryId)ref.getParent());
+
+		if (gitInfo == null) {
+            try (InputStream is = Repository.INSTANCE.newInputStream(ref)) {
+                if (is != null) {
+                    ArchiveEntry archiveEntry = new ZipArchiveEntry(archivePath);
+                    zos.putArchiveEntry(archiveEntry);
+                    IOUtils.copy(is, zos);
+					zos.closeArchiveEntry();
+                }
+            } catch (Exception e) {
+                CSARExporter.LOGGER.error("Could not copy file to ZIP outputstream", e);
+            }
+			return;
+        }
+
+		Path tempDir = Files.createTempDirectory(WINERY_TEMP_DIR_PREFIX);
+		try {
+            Git git = Git
+                    .cloneRepository()
+                    .setURI(gitInfo.URL)
+                    .setDirectory(tempDir.toFile())
+                    .call();
+            git.checkout().setName(gitInfo.BRANCH).call();
+            String path = "artifacttemplates/"
+                    + Util.URLencode(((ArtifactTemplateId)ref.getParent().getParent()).getQName().getNamespaceURI())
+                    + "/"
+                    + ((ArtifactTemplateId)ref.getParent().getParent()).getQName().getLocalPart()
+                    + "/files/";
+			TArtifactTemplate template = Utils.getTArtifactTemplate((ArtifactTemplateDirectoryId) ref.getParent());
+			addWorkingTreeToArchive(zos, template, tempDir, path);
+        } catch (GitAPIException e) {
+            CSARExporter.LOGGER.error(String.format("Error while cloning repo: %s / %s", gitInfo.URL, gitInfo.BRANCH), e);
+        } finally {
+            deleteDirectory(tempDir);
+        }
+	}
+
+	/**
+	 * Adds a file to an archive
+	 * @param zos Output stream of the archive
+	 * @param ref Reference to the file that should be added to the archive
+	 * @param archivePath Path inside the archive to the file
+	 */
+	private void addFileToZipArchive(ArchiveOutputStream zos, RepositoryFileReference ref, String archivePath) {
+		try (InputStream is = Repository.INSTANCE.newInputStream(ref)) {
+            ArchiveEntry archiveEntry = new ZipArchiveEntry(archivePath);
+            zos.putArchiveEntry(archiveEntry);
+            IOUtils.copy(is, zos);
+        } catch (Exception e) {
+            CSARExporter.LOGGER.error("Could not copy file content to ZIP outputstream", e);
+        }
+	}
+
+	/**
```

Comment:
I didn't create the code, I just extracted the method. I tried to add comments as good as possible.